### ISSUE TITLE
Heap-use-after-free src/jsiValue.c:215 in Jsi_ValueReset

### ISSUE THREAD

**Issue by hope-fly** [pcmacdon/jsish#75]
###### Jsish revision

Commit: [9fa798e](https://github.com/pcmacdon/jsish/commit/9fa798e35d7a753672d1bd16e98b124e7b2d2899)

Version: v3.5.0

###### Build platform

Ubuntu 18.04.5 LTS (Linux 5.4.0-44-generic x86_64)

###### Build steps

```bash
export CFLAGS='-fsanitize=address'
make
```

###### Test case

```javascript
var JSEtest = [
  'aa',
  'hhh',
  'bb'
];
'sort:' + JSEtest.sort(function (str, position) {
  return JSEtest.unshift('pop:' + JSEtest.pop());
});
!'concat:' + JSEtest.concat().every(function (E) {
  return JSEtest.pop(),
  'mm',
  'nn';
});
```

​	

###### Execution steps & Output

```bash
$ ./jsish/jsish poc.js
==75939==ERROR: AddressSanitizer: heap-use-after-free on address 0x603000006fd0 at pc 0x5579bee3dabb bp 0x7ffc7f6b2940 sp 0x7ffc7f6b2930
READ of size 4 at 0x603000006fd0 thread T0
    #0 0x5579bee3daba in Jsi_ValueReset src/jsiValue.c:215
    #1 0x5579bf179269 in jsiClearStack src/jsiEval.c:123
    #2 0x5579bf179269 in jsiPop src/jsiEval.c:200
    #3 0x5579bf179269 in jsiEvalCodeSub src/jsiEval.c:1390
    #4 0x5579bf19515e in jsi_evalcode src/jsiEval.c:2204
    #5 0x5579beeb8834 in jsi_FuncCallSub src/jsiProto.c:220
    #6 0x5579bee34fec in jsi_FunctionInvoke src/jsiFunc.c:777
    #7 0x5579bee34fec in Jsi_FunctionInvoke src/jsiFunc.c:789
    #8 0x5579beefbf64 in jsi_ArrayFindSubCmd src/jsiArray.c:576
    #9 0x5579beefbf64 in jsi_ArrayEveryCmd src/jsiArray.c:663
    #10 0x5579beeb7818 in jsi_FuncCallSub src/jsiProto.c:244
    #11 0x5579bf18171a in jsiFunctionSubCall src/jsiEval.c:796
    #12 0x5579bf18171a in jsiEvalFunction src/jsiEval.c:837
    #13 0x5579bf18171a in jsiEvalCodeSub src/jsiEval.c:1264
    #14 0x5579bf19515e in jsi_evalcode src/jsiEval.c:2204
    #15 0x5579bf199274 in jsi_evalStrFile src/jsiEval.c:2665
    #16 0x5579bee8866a in Jsi_Main src/jsiInterp.c:936
    #17 0x5579bf68d03a in jsi_main src/main.c:47
    #18 0x7febc1aa1bf6 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21bf6)
    #19 0x5579bee1c969 in _start (/usr/local/bin/jsish+0xe8969)

0x603000006fd0 is located 0 bytes inside of 32-byte region [0x603000006fd0,0x603000006ff0)
freed by thread T0 here:
    #0 0x7febc27107a8 in __interceptor_free (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xde7a8)
    #1 0x5579bee3d6cf in Jsi_DecrRefCount src/jsiValue.c:52

previously allocated by thread T0 here:
    #0 0x7febc2710d28 in __interceptor_calloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xded28)
    #1 0x5579bee8daa4 in Jsi_Calloc src/jsiUtils.c:57

SUMMARY: AddressSanitizer: heap-use-after-free src/jsiValue.c:215 in Jsi_ValueReset
Shadow bytes around the buggy address:
  0x0c067fff8da0: fd fd fd fd fa fa fd fd fd fd fa fa fd fd fd fd
  0x0c067fff8db0: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fd fd
  0x0c067fff8dc0: fd fd fa fa 00 00 00 00 fa fa 00 00 00 00 fa fa
  0x0c067fff8dd0: 00 00 00 00 fa fa 00 00 00 00 fa fa 00 00 00 00
  0x0c067fff8de0: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fd fd
=>0x0c067fff8df0: fd fd fa fa fd fd fd fd fa fa[fd]fd fd fd fa fa
  0x0c067fff8e00: fd fd fd fd fa fa fd fd fd fd fa fa fd fd fd fd
  0x0c067fff8e10: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fd fd
  0x0c067fff8e20: fd fd fa fa fd fd fd fd fa fa fd fd fd fd fa fa
  0x0c067fff8e30: fd fd fd fd fa fa 00 00 00 00 fa fa fd fd fd fd
  0x0c067fff8e40: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fd fd
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==75939==ABORTING

```

**Comment by pcmacdon** [comment]
Duplicate bug.

**Comment by pcmacdon** [comment]
Fixed in #85
